modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js

```diff
@@ -54,6 +54,17 @@ export default class Tile3DLayer extends CompositeLayer {
     }
   }
 
+  getPickingInfo({info, sourceLayer}) {
+    const {layerMap} = this.state;
+    const layerId = sourceLayer && sourceLayer.id;
+    if (layerId) {
+      const tileId = layerId.substring(this.id.length + 1).split('-')[1];
```

Comment:
What if `tileId` from some new loader contains `-`?

Reply:
this should be resolved inside deck.gl `tile-3d-layer`, a better naming could be suggested. 

https://github.com/uber/deck.gl/blob/master/modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js#L182

Reply:
If `tile.id` contains `-` you will end up with a partial id that does not match any thing. You need to take the rest of the string from the first `-`.

Reply:
layer id has three parts, `${this.id}-scenegraph-${tileHeader.fullUri}`

`layerId.substring(this.id.length + 1)` will give `scenegraph-${tileHeader.fullUri}`

`scenegraph-${tileHeader.fullUri}`.split('-')[1]` will give me the rest of the string after the `${tileHeader.fullUri}`.